src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/PopupRenderer.ios.cs

```diff
@@ -0,0 +1,288 @@
+					case LayoutAlignment.End:
+						originX = UIScreen.MainScreen.Bounds.Width;
+						break;
+					case LayoutAlignment.Center:
+						originX = UIScreen.MainScreen.Bounds.Width / 2;
+						break;
+				}
+
+				PopoverPresentationController.SourceRect = new CGRect(originX, originY, 0, 0);
+			}
+			else
+			{
+				var view = Platform.GetRenderer(Element.Anchor).NativeView;
+				PopoverPresentationController.SourceView = view;
+				PopoverPresentationController.SourceRect = view.Bounds;
+			}
+		}
+
+		void SetBackgroundColor()
+		{
+			Control.NativeView.BackgroundColor = Element.Color.ToUIColor();
+		}
+
+		void SetView()
+		{
+			base.View.AddSubview(Control.ViewController.View);
+			base.View.Bounds = new CGRect(0, 0, PreferredContentSize.Width, PreferredContentSize.Height);
+			AddChildViewController(Control.ViewController);
+		}
+
+		void SetPresentationController()
+		{
+			((UIPopoverPresentationController)PresentationController).SourceView = ViewController.View;
+
+			// Setting PermittedArrowDirector to 0 breaks the Popover layout. It would be nice if there is no anchor to remove the arrow.
+			((UIPopoverPresentationController)PresentationController).PermittedArrowDirections = UIPopoverArrowDirection.Up;
+			((UIPopoverPresentationController)PresentationController).Delegate = new PopoverDelegate(this);
+		}
+
+		void AddToCurrentPageViewController()
+		{
+			ViewController.PresentViewController(this, true, () => Element.OnOpened());
+		}
+
+		async void OnDismissed(object sender, PopupDismissedEventArgs e)
+		{
+			if (!FormsToolkit.IsiOS9OrNewer)
+				await ViewController.DismissViewControllerAsync(true);
+			else
+				ViewController.DismissViewController(true, null);
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (isDisposed)
+				return;
+
+			isDisposed = true;
+			if (disposing)
+			{
+				if (Element != null)
+				{
+					Element.PropertyChanged -= OnElementPropertyChanged;
+
+					if (Platform.GetRenderer(Element) == this)
+					{
+						// NOTE - AH 9/12/2020
+						// This used to use the internal property
+						// 'Xamarin.Forms.Platform.Android.Platform.RendererProperty'
+						// That property is marked internal so the closest thing we can do
+						// is duplicate the implementation here.
+						//
+						// I don't think this is really needed for the control, but I am
+						// leaving it in so it can be reviewed.
+						var rendererProperty = BindableProperty.CreateAttached("Renderer", typeof(IVisualElementRenderer), typeof(Platform), default(IVisualElementRenderer), propertyChanged: (bindable, oldvalue, newvalue) =>
+						{
+							var view = bindable as VisualElement;
+							if (view != null)
+								view.IsPlatformEnabled = newvalue != null;
+						});
+
+						Element.ClearValue(rendererProperty);
+					}
+
+					Element = null;
+				}
+			}
+
+			base.Dispose(disposing);
+		}
+
+		class PopoverDelegate : UIPopoverPresentationControllerDelegate
+		{
+			PopupRenderer popupRenderer;
```

Comment:
We are passing a strong reference to this delegate, I'm not sure if the Xamarin + GC, can handle that or if we need to dispose it. What do you think to use a `WeakReference` here?

Reply:
Do you have the docs to `WeakReference`? I will take a look and see if it works

Reply:
You can see [this](https://github.com/brminnick/AsyncAwaitBestPractices#weakeventmanager), our implementation was ported from there.

Reply:
That makes sense, I thought `WeakReference` was another technique. I'll use the `WeakEventManager` to allow the Delegate to communicate with the renderer. This is a good idea to ensure the GC cleans up unused memory.

Reply:
I took a look at how the Xamarin.Forms `WeakEventManager` works and it appeared that I still needed to pass a reference of the `WeakEventManager` to the delegate class. I'll include the code in the next update, but have also included the snippets below. Not 100% sure if this is what you had in mind.

```c#
readonly WeakEventManager eventManager = new WeakEventManager();
public PopupRenderer()
{
    eventManager.AddEventHandler(OnDismiss, nameof(OnDismiss));
}

void OnDimiss(object sender, EventArgs args)
{
    if (IsViewLoaded && Element.IsLightDismissEnabled)
        Element.LightDismiss();
}

class PopoverDelegate : UIPopoverPresentationControllerDelegate
{
	readonly WeakEventManager eventManager;

	public PopoverDelegate(WeakEventManager eventManager) =>
		this.eventManager = eventManager;

	public override UIModalPresentationStyle GetAdaptivePresentationStyle(UIPresentationController forPresentationController) =>
		UIModalPresentationStyle.None;

	public override void DidDismiss(UIPresentationController presentationController) =>
		eventManager.RaiseEvent(this, new EventArgs(), nameof(OnDismiss));
}
```

My confusion is we are still passing an object from the renderer->delegate. Not sure if this solves the problem.

Reply:
@brminnick you are more familiar with WeakEventManager. Does this make sense? Do you have any tips?

Reply:
This is not the correct way to use `WeakEventManager`